### PR TITLE
Add a copy operation to the Document class

### PR DESCRIPTION
**Pull request by mikerhodes**
Thanks for your hard work, please ensure all items are complete before opening.

- [x] Tick to sign-off your agreement to the [Developer Certificate of Origin (DCO) 1.1](https://github.com/cloudant/python-cloudant/blob/master/DCO1.1.txt)
- [x] You have added tests for any code changes
- [x] You have updated the [CHANGES.rst](https://github.com/cloudant/python-cloudant/blob/master/CHANGES.rst) 
- [x] You have completed the PR template below:

## What

Added support for (shallow) copying Document objects.

## How

Added a public `copy` method and private `__copy__` method. This means that a document can be copied via a call to the object itself and also via the `copy` module, that is `copy.copy(doc)`. The latter allows for calling code to be more generic.

## Testing

I added a test that both `copy` and `copy.copy` produce accurate copies and that they are distinct objects.

## Issues

N/A, minor change.

### CODE REVIEW

File: src/cloudant/document.py [hunk 1]
```diff
@@ -65,6 +65,21 @@ def __init__(self, database, document_id=None):
...
+        """
+        return self.__copy__()
+
+    def __copy__(self):
+        """
+        Copy the document, for use in copy.copy() calls. See `copy()`.
+        """
+        cpy = Document(self._database, document_id=self._document_id)
+        for k, v in self.items():
+            cpy[k] = v
```
Comment (smithsz) [inline src/cloudant/document.py]: Could `v` be a mutable type here? If so it'll get passed by reference and you won't get a deep copy.

Comment (mikerhodes) [reply src/cloudant/document.py]: I don't intend for it to be a deep copy, only a shallow copy. See "What", "Added support for (shallow) copying Document objects." :) 

Whether this is the right intent we can discuss, but I took the normal dict copy behaviour.

**Comment by smithsz** [comment]
Unfortunately this doesn't work for subclasses of `Document`:
```python
(Pdb) from cloudant.design_document import DesignDocument
(Pdb) ddoc1 = DesignDocument(self.db, "myddoc")
(Pdb) ddoc2 = copy.deepcopy(ddoc1)
(Pdb) ddoc2.__class__
<class 'cloudant.document.Document'>  # should be class DesignDocument
```
What's the issue with calling `copy.copy(mydoc)` or `copy.deepcopy(mydoc)` without specifying these custom implementations?

**Comment by mikerhodes** [comment]
Seems like, as you say, `copy.copy` and `copy.deepcopy` work fine. Oh well.

**Pull request closed without merging**